src/zkp/pifac.rs

```diff
@@ -329,136 +332,143 @@ fn sqrt(num: &BigNumber) -> BigNumber {
-                &input,
-                &ProverSecret::new(&not_p0, &not_q0),
+                input,
+                ProverSecret::new(&not_p0, &not_q0),
                 &(),
-                &mut transcript,
-                &mut rng,
+                &mut transcript(),
+                &mut rng2,
             )?;
-            let mut transcript = Transcript::new(b"PiFac Test");
+
             assert!(incorrect_factors
-                .verify(&input, &(), &mut transcript)
+                .verify(input, &(), &mut transcript())
                 .is_err());
 
-            let mut transcript = Transcript::new(b"PiFac Test");
             let small_p = BigNumber::from(7u64);
             let small_q = BigNumber::from(11u64);
-            let setup_params = VerifiedRingPedersen::gen(&mut rng, &())?;
-            let small_input = CommonInput::new(&setup_params, &(&small_p * &small_q));
+            let setup_params = VerifiedRingPedersen::gen(&mut rng2, &())?;
+            let modulus = &small_p * &small_q;
+            let small_input = CommonInput::new(&setup_params, &modulus);
             let small_proof = PiFacProof::prove(
-                &input,
-                &ProverSecret::new(&small_p, &small_q),
+                input,
+                ProverSecret::new(&small_p, &small_q),
                 &(),
-                &mut transcript,
-                &mut rng,
+                &mut transcript(),
+                &mut rng2,
             )?;
-            let mut transcript = Transcript::new(b"PiFac Test");
+
             assert!(small_proof
-                .verify(&small_input, &(), &mut transcript)
+                .verify(small_input, &(), &mut transcript())
                 .is_err());
 
-            let mut transcript = Transcript::new(b"PiFac Test");
-            let regular_sized_q = prime_gen::try_get_prime_from_pool_insecure(&mut rng).unwrap();
-            let mixed_input = CommonInput::new(&setup_params, &(&small_p * &regular_sized_q));
+            let regular_sized_q = prime_gen::try_get_prime_from_pool_insecure(&mut rng2).unwrap();
+            let modulus = &small_p * &regular_sized_q;
+            let mixed_input = CommonInput::new(&setup_params, &modulus);
             let mixed_proof = PiFacProof::prove(
-                &input,
-                &ProverSecret::new(&small_p, &regular_sized_q),
+                input,
+                ProverSecret::new(&small_p, &regular_sized_q),
                 &(),
-                &mut transcript,
-                &mut rng,
+                &mut transcript(),
+                &mut rng2,
             )?;
-            let mut transcript = Transcript::new(b"PiFac Test");
+
             assert!(mixed_proof
-                .verify(&mixed_input, &(), &mut transcript)
+                .verify(mixed_input, &(), &mut transcript())
                 .is_err());
 
-            let mut transcript = Transcript::new(b"PiFac Test");
             let small_fac_p = &not_p0 * &BigNumber::from(2u64);
```

Comment:
Todo: add docs here:" modulus cannot have small factors".

For this particular test, I would rather put it in its own function that does not call the with-random-small-factors-proof function because I think it does not use any of the fields provided by that.